core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -658,6 +658,17 @@ class MapUnit {
             .forEach { unit -> unit.destroy() }
     }
 
+    fun gift(recipient: CivilizationInfo) {
+        civInfo.removeUnit(this)
+        civInfo.updateViewableTiles()
+        // all transported units should be destroyed as well
+        currentTile.getUnits().filter { it.isTransported && isTransportTypeOf(it) }
```

Comment:
IIRC unit.destroy() does this internally, no?

Reply:
I definitely recall this being somewhere. Also that air units try to move to closest city if they still have movement points so that they aren't destroyed.

Reply:
It does, but also removeFromTile(), which we don't want